### PR TITLE
Handle error rate

### PR DESCRIPTION
**Pull request by apregier**
This pr is a little strange - I am fixing how the alignment stats summary prints the error rate when it is not available (it should print 'NA'), while at the same time causing the process diff to ignore that column.  The reason for ignoring it is that we don't have a way to make a blessed process for the cle regression test in production that would have 'NA' in this column - because the value is available.  However, in our clean test db that is used for the test, this value will not be available and should be NA (this is actually what is expected for most cle data - the error rate will not be available in this report).
If/when we update the cle processing profile to produce error rate information, and update the alignment stats report to access the error rate information from a place that will actually be produced in the cle, we can remove this exception to the diff.

### CODE REVIEW

File: lib/perl/Genome/Model/SomaticValidation/Command/AlignmentStatsSummary.pm [hunk 1]
```diff
@@ -144,7 +144,8 @@ sub _alignment_metrics_from_result {
         $total_bases += $lane->total_base_count;
         $total_mapped_bases += $lane->total_aligned_base_count;
         my $filt_error_rate_avg;
-        if ($mismatches ne "NA" and $lane->instrument_data->can("filt_error_rate_avg")) {
+        if ($mismatches ne "NA" and $lane->instrument_data->can("filt_error_rate_avg")
+            and defined($lane->instrument_data->can("filt_error_rate_avg"))) {
```
Comment (tmooney) [inline lib/perl/Genome/Model/SomaticValidation/Command/AlignmentStatsSummary.pm]: This last one should call `filt_error_rate_avg` instead of `can`.

Comment (apregier) [inline lib/perl/Genome/Model/SomaticValidation/Command/AlignmentStatsSummary.pm]: Thanks for the catch - awesome to be introducing bugs and removing the test that would potentially catch them at the same time :-\

**Comment by acoffman** [comment]
+1

**Comment by tmooney** [comment]
+1 :camel:

**Comment by ernfrid** [comment]
@apregier - should we rethink the strategy here on simply ignoring these columns in the alignment summary? Your comment about removing tests make me much less comfortable with this approach than I was this morning.

**Comment by apregier** [comment]
The right thing to do may be to write a unit test - then we are not depending on the cle test for this particular detail.

**Comment by apregier** [comment]
@ernfrid I just pushed a test case that failed before the change (it showed an error rate of 0.00) and now passes, showing 'NA'.  Does that make you feel better?

**Comment by apregier** [comment]
fwiw it makes _me_ feel better

**Comment by ernfrid** [comment]
Yay! Makes me feel better too.

**Pull request merged**